src/component_types/character.lua

```diff
@@ -60,76 +60,95 @@ return {
         this.squish        = 0
         this.squishiness   = this.squishiness or 1
         this.squishspeed   = 32 -- pixels per second
+
+        this.nade_xoffset = this.nade_xoffset or 0
+        this.nade_yoffset = this.nade_yoffset or 0
     end,
 
     explode = function(this, _, _, _)
-        this.dead = true
-
-        local sprite_left = this.x + this.spr_offsetx
-
-        this.follow_gib = object_group.create_object(this.__layer, 'gib', {
-            spr_name = '12x9_player_head.png',
-            x = sprite_left + 11,
-            y = this.y,
-            dx = this.dx,
-            dy = this.dy,
-            flip = (this.direction == 'left'),
-            color = this.color,
-        })
-
-        object_group.create_object(this.__layer, 'gib', {
-            spr_name = '14x13_player_body.png',
-            x = sprite_left + 8,
-            y = this.y + 8,
-            dx = this.dx,
-            dy = this.dy,
-            color = this.color,
-        })
-
-        object_group.create_object(this.__layer, 'gib', {
-            spr_name = '5x9_player_leg.png',
-            x = sprite_left + 14,
-            y = this.y + 22,
-            dx = this.dx,
-            dy = this.dy,
-            color = this.color,
-        })
-
-        object_group.create_object(this.__layer, 'gib', {
-            spr_name = '5x9_player_leg.png',
-            x = sprite_left + 18,
-            y = this.y + 22,
-            dx = this.dx,
-            dy = this.dy,
-            color = this.color,
-        })
-
-        object_group.create_object(this.__layer, 'gib', {
-            spr_name = '6x13_player_arm.png',
-            x = sprite_left + 8,
-            y = this.y + 8,
-            dx = this.dx,
-            dy = this.dy,
-            color = this.color,
-        })
-
-        object_group.create_object(this.__layer, 'gib', {
-            spr_name = '6x13_player_arm.png',
-            x = sprite_left + 18,
-            y = this.y + 8,
-            dx = this.dx,
-            dy = this.dy,
-            color = this.color,
-        })
+        -- if we're the player
+        -- and we're going to explode
+        -- don't die, cuz that's lame
+        -- A haiku by quigley-c
+        if this.__parent == 'player' then
```

Comment:
`this.__parent` will refer to an object, which will be a table type; `this.__parent.__typename == 'player'` should work here.